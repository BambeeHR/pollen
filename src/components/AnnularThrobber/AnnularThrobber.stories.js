import AnnularThrobber, { Sizes } from './AnnularThrobber.vue';

export default {
  title: 'AnnularThrobber',
  component: AnnularThrobber,
};

export const Default = () => ({
  components: { AnnularThrobber },
  template: `<AnnularThrobber />`,
});

export const Gallery = () => ({
  components: { AnnularThrobber },
  data() {
    return {
      sizes: Object.values(Sizes),
    };
  },
  template: `
    <div class="flex">
      <div v-for="size in sizes" :key="size" class="flex flex-col justify-between mr-4">
        <div class="mb-2">{{ size }}</div>
        <AnnularThrobber :size="size" />
      </div>
    </div>
  `,
});
