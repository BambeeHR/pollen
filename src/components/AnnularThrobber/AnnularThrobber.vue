<template functional>
  <div
    class="annular-throbber"
    :class="[`annular-throbber--${props.size}`, data.class, data.staticClass]"
    v-bind="$options.getAttributes({ data, parent })"
    v-on="listeners"
  />
</template>

<script>
import getAttributes from '../../util/getAttributes';

export const Sizes = Object.freeze({
  SM: 'sm',
  MD: 'md',
  LG: 'lg',
  XL: 'xl',
});

/**
 * A ring-shaped graphical element to show that an action is in progress. In
 * contrast to a progress bar, a throbber does not convey how much of the action
 * has been completed.
 */
export default {
  getAttributes,
  props: {
    /** One of `sm`, `md`, `lg`, or `xl`. */
    size: {
      type: String,
      default: Sizes.MD,
      validator: (value) => Object.values(Sizes).includes(value),
    },
  },
};
</script>

<style scoped>
.annular-throbber {
  animation-duration: 2s;
  animation-iteration-count: infinite;
  animation-timing-function: steps(60);
  background-size: auto 100%;
}

@keyframes moveSm {
  from {
    background-position-x: 0;
  }
  to {
    background-position-x: -3840px;
  }
}

.annular-throbber--sm {
  @apply h-8 w-8;
  animation-name: moveSm;
  background-image: url(https://images.ctfassets.net/tb3mx4vb7wqk/7ly72lwgXSOmq29YG21AXt/09f7d806694add0cd3bbd79d6e388a44/logo-sprite.png?h=64);
}

@keyframes moveMd {
  from {
    background-position-x: 0;
  }
  to {
    background-position-x: -5760px;
  }
}

.annular-throbber--md {
  @apply h-12 w-12;
  animation-name: moveMd;
  background-image: url(https://images.ctfassets.net/tb3mx4vb7wqk/7ly72lwgXSOmq29YG21AXt/09f7d806694add0cd3bbd79d6e388a44/logo-sprite.png?h=96);
}

@keyframes moveLg {
  from {
    background-position-x: 0;
  }
  to {
    background-position-x: -7680px;
  }
}

.annular-throbber--lg {
  @apply h-16 w-16;
  animation-name: moveLg;
  background-image: url(https://images.ctfassets.net/tb3mx4vb7wqk/7ly72lwgXSOmq29YG21AXt/09f7d806694add0cd3bbd79d6e388a44/logo-sprite.png?h=128);
}

@keyframes moveXl {
  from {
    background-position-x: 0;
  }
  to {
    background-position-x: -11520px;
  }
}

.annular-throbber--xl {
  @apply h-24 w-24;
  animation-name: moveXl;
  background-image: url(https://images.ctfassets.net/tb3mx4vb7wqk/7ly72lwgXSOmq29YG21AXt/09f7d806694add0cd3bbd79d6e388a44/logo-sprite.png?h=192);
}
</style>
