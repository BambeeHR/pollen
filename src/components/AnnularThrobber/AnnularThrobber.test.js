import { mount } from '@vue/test-utils';
import AnnularThrobber from './AnnularThrobber.vue';
import { Gallery } from './AnnularThrobber.stories';

describe('LoadingSprite', () => {
  test('renders correctly', () => {
    const wrapper = mount(AnnularThrobber);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders gallery correctly', () => {
    const gallery = Gallery();
    const wrapper = mount(gallery);
    expect(wrapper.element).toMatchSnapshot();
  });
});
