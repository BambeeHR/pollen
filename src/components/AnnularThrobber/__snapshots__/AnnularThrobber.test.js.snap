// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSprite renders correctly 1`] = `
<div
  class="annular-throbber annular-throbber--md"
/>
`;

exports[`LoadingSprite renders gallery correctly 1`] = `
<div
  class="flex"
>
  <div
    class="flex flex-col justify-between mr-4"
  >
    <div
      class="mb-2"
    >
      sm
    </div>
     
    <div
      class="annular-throbber annular-throbber--sm"
    />
  </div>
  <div
    class="flex flex-col justify-between mr-4"
  >
    <div
      class="mb-2"
    >
      md
    </div>
     
    <div
      class="annular-throbber annular-throbber--md"
    />
  </div>
  <div
    class="flex flex-col justify-between mr-4"
  >
    <div
      class="mb-2"
    >
      lg
    </div>
     
    <div
      class="annular-throbber annular-throbber--lg"
    />
  </div>
  <div
    class="flex flex-col justify-between mr-4"
  >
    <div
      class="mb-2"
    >
      xl
    </div>
     
    <div
      class="annular-throbber annular-throbber--xl"
    />
  </div>
</div>
`;
